src/store/reducers/MovieReducer.js

```diff
@@ -86,6 +101,54 @@ const movieReducer = (state = initialState, action) => {
     }
     case UPDATE_MOVIE_VISITS:
       return { ...state, currentMovie: {...state.currentMovie, visits: state.currentMovie.visits + 1 } };
+    case PUT_MOVIE_COMMENT: {
+      console.log({
+          ...state.currentMovie,
+          comments: {
+            ...state.currentMovie.comments,
+            results: [
+              action.payload,
+              ...state.currentMovie.comments.results.filter((result, index, self) => 
+                self.findIndex(r => result.timestamp === r.timestamp) === index
+              )
+            ],
+            total: state.currentMovie.comments.total + 1
+          }
+        } 
+      )
+      return { 
+        ...state, 
+        currentMovie: {
+          ...state.currentMovie,
+          comments: {
+            ...state.currentMovie.comments,
+            results: [
+              action.payload,
+              ...state.currentMovie.comments.results.filter((result, index, self) => 
+                self.findIndex(r => result.timestamp === r.timestamp) === index
```

Comment:
cemu sliz ovaj timestamp ? i pordejenje sa index-om ?

Reply:
Ovo koristim zbog nekih specijalnih slucajevima u kojima se mogu pojaviti duplikati za neke komentare. Recimo da ovo radi disctinct nad listom komentara. 
Na primer prikazano je 10 komentara i sada korisnik doda jos jedan komentar i sada je prikazano 11 komentara. Posto mi je page size za paging komentara 10, u tom slucaju bi doslo do toga da poslednji komentar od tih 10 koji su prvobitno prikazani predje na page 2 i kada bih odradio load more funkciju on bi se ucitao u okviru te druge stranice, a ovamo vec postoji prikazan.

Veoma sam detaljno o svemu tome razmisljao, nezgodno je to refresh-ovanje state-a u ovom konkretnom slucaju. Ali sam na ovaj nacin nasao kompromis i bazirao se na tome da korisniku ne prikazujem nijedan komentar duplo.